docs/source/deployment/deploy-model-locally.rst

```diff
@@ -0,0 +1,245 @@
+This command starts a local server that listens on the specified port and serves your model.
+
+.. tabs::
+
+    .. code-tab:: bash
+
+       mlflow models serve -m runs:/<run_id>/model -p 5000
+
+    .. code-tab:: python
+
+       import mlflow
+
+       model = mlflow.pyfunc.load_model("runs:/<run_id>/model")
+       model.serve(port=5000)
+
+
+You can then send a test request to the server as follows:
+
+.. code-block:: bash
+
+    curl http://127.0.0.1:5000/invocations -H "Content-Type:application/json;"  --data '{"inputs": [[1, 2], [3, 4], [5, 6]]}'
+
+
+Several command line options are available to customize the server's behavior. For instance, the ``--env-manager`` option allows you to
+choose a specific environment manager, like Anaconda, to create the virtual environment. The `mlflow models` module also provides
+additional useful commands, such as building a Docker image or generating a Dockerfile. For comprehensive details, please refer 
+to the `MLflow CLI Reference <../cli.html#mlflow-models>`_.
+
+
+.. _local-inference-server-spec:
+
+Inference Server Specification
+------------------------------
+
+Endpoints
+~~~~~~~~~
+The inference server provides 4 endpoints:
+
+* ``/invocations``: An inference endpoint that accepts POST requests with input data and returns predictions.
+
+* ``/ping``: Used for health checks.
+
+* ``/health``: Same as /ping
+
+* ``/version``: Returns the MLflow version.
+
+Accepted Input Formats
+~~~~~~~~~~~~~~~~~~~~~~
+The ``/invocations`` endpoint accepts CSV or JSON inputs. The input format must be specified in the
+``Content-Type`` header as either ``application/json`` or ``application/csv``.
+
+CSV Input
+*********
+
+CSV input must be a valid pandas.DataFrame CSV representation. For example:
+
+``curl http://127.0.0.1:5000/invocations -H 'Content-Type: application/csv' --data '1,2,3,4'``
+
+JSON Input
+**********
+
+JSON input must be a dictionary with exactly one of the following fields, further specifying the input data's type and encoding:
+
+.. list-table::
+    :widths: 20 40 40
+    :header-rows: 1
+    :class: wrap-table
+
+    * - Field
+      - Description
+      - Example
+    * - ``dataframe_split``
+      - Pandas DataFrames in the ``split`` orientation.
+      - ``{"dataframe_split": pandas_df.to_dict(orient='split')``
+    * - ``dataframe_records``
+      - Pandas DataFrame in the ``records`` orientation. **We do not recommend using this format because it is not guaranteed to preserve column ordering.**
+      - ``{"dataframe_records": pandas_df.to_dict(orient='records')``
+    * - ``instances``
+      - Tensor input formatted as described in `TF Serving's API docs <https://www.tensorflow.org/tfx/serving/api_rest#request_format_2>`_ where the provided inputs will be cast to Numpy arrays.
+      - ``{"instances": [1.0, 2.0, 5.0]}``
+    * - ``inputs``
+      - Same as ``instances`` but with a different key.
+      - ``{"inputs": [1.0, 2.0, 5.0]}``
```

Comment:
Could we use a List[str] example instead?

Reply:
Sure! I guess because this supposed to be used for LLM case...?